pymongo/results.py

```diff
@@ -28,15 +28,15 @@ class _WriteResult:
 
     __slots__ = ("__acknowledged",)
 
-    def __init__(self, acknowledged: bool) -> None:
+    def __init__(self, acknowledged: Optional[bool] = None) -> None:
```

Comment:
Why is acknowledged optional now? What does that case mean?

Reply:
`acknowledged` is None for an individual write result within the ClientBulkWriteResult, since the spec mentions that acknowledgement info is not relevant for individual ops within the bulk write. Would it make more sense to pass `acknowledged=True` in these cases?

Reply:
If the ClientBulkWriteResult is acknowledged=False, then there are no results to inspect at all. If the ClientBulkWriteResult is acknowledged, the all the children results should also be acknowledged=True. Does that sound correct?

Reply:
Yep that sounds correct!